CDTS Blazor/Data/Services/AzureBlobService.cs

```diff
@@ -20,41 +22,45 @@ public AzureBlobService(IAzureBlobConnectionFactory azureBlobConnectionFactory)
 
         public async Task<CloudBlockBlob> UploadFileAsync(IFormFile file, string container = null)
         {
+            // Perhaps we can fail more gracefully then just throwing an exception
+            if (file == null)
+                throw new ArgumentNullException(nameof(file));
+            
+            var blobContainer = await _azureBlobConnectionFactory.GetBlobContainer().ConfigureAwait(false);
 
-            var blobContainer = await _azureBlobConnectionFactory.GetBlobContainer(container);
-
-            var blobName = UniqueFileName(file.FileName);
+            var blobName = AzureBlobService.UniqueFileName(file.FileName);
 
             // Create the blob to hold the data
             var blob = blobContainer.GetBlockBlobReference(blobName);
             // Send the file to the cloud storage
             using (var stream = file.OpenReadStream())
             {
-                await blob.UploadFromStreamAsync(stream);
+                await blob.UploadFromStreamAsync(stream).ConfigureAwait(false);
             }
 
             return blob;
-
         }
 
         public async Task<List<CloudBlockBlob>> UploadMultipleFilesAsync(IFormFileCollection files, string container = null)
-
         {
+            // Perhaps we can fail more gracefully then just throwing an exception
```

Comment:
Wouldn't that be up to the UI?

Reply:
That's a good point we definitely need a way of surfacing those up to the UI in some way.